### DISCUSSION TITLE
Compute Module 4 - two RPI camera

### DISCUSSION THREAD

**Discussion by projectile25** [Motion-Project/motion#1432] [Q&A]
Hi , is some possibility to use two Rpi cameras on CM4 trough the mmal ?  They both work like v4l2  devices or one like v4l2 and second trought the mmal .  But when i setup both like mmal, the only one camera is working , the second has error message  "MMAL cam failed to open" .  I have setup in config for each camera the mmalcam_name vc.ril.camera and  mmal_params -cs 0 and mmal_params -cs 1 .

**Comment by tosiara** [comment]
I don't own a pi camera and have never seen anyone posting issues about using more than one camera.
Can you upload a detailed log file `-d 9` with two mmal cameras failing?

**Reply by projectile25** [reply]
Hi 
attached log .Thank you 

[motion.log](https://github.com/Motion-Project/motion/files/7539255/motion.log)

**Comment by tosiara** [comment]
maybe @jogu (he made the original MMAL pull request) could comment here: does MMAL support running two threads with the same `mmalcam_name vc.ril.camera` specified?

**Reply by jogu** [reply]
I didn't do too much beyond take someone else's working MMAL code and making it work against the latest motion version so it could be merged in.

I'd be highly surprised if it was able to support two MMAL cameras as the Pi hardware only supported one camera at the time. I'd presume (for someone with access to the relevant hardware) it's probably not too tricky to add support for both cameras though.

**Reply by projectile25** [reply]
Hi, if it would help anything , here is motion log for two rpi cameras working via V4l2 . 

[motion_v4l2.log](https://github.com/Motion-Project/motion/files/7540705/motion_v4l2.log)
